src/core/lib/channel/http_client_filter.h

```diff
@@ -41,4 +41,8 @@ extern const grpc_channel_filter grpc_http_client_filter;
 /* Channel arg to override the http2 :scheme header */
 #define GRPC_ARG_HTTP2_SCHEME "grpc.http2_scheme"
 
+/* Channel arg to determine maximum size of payload eligable for GET request */
+#define GRPC_ARG_MAX_PAYLOAD_SIZE_FOR_GET "grpc.max_payload_size_for_get"
+static const size_t kMaxPayloadSizeForGet = 2048;
```

Comment:
We need to expose this channel arg. It is the only way a client app can prevent GET requests at run-time.